.buildkite/pipeline_cpu_template.py

```diff
@@ -0,0 +1,167 @@
+
+import argparse
+import json
+from enum import Enum
+
+from common import group
+
+class bkstep(str, Enum):
+    LABEL="label"
+    TIMEOUT="timeout"
+    COMMAND="commands"
+    ARTIFACTS="artifact_paths"
+
+cpu_template_test = {
+    "rdmsr": {
+        bkstep.COMMAND: [
+            "tools/devtool -y test -- \
+            -s -ra --nonci --log-cli-level=INFO \
+            integration_tests/functional/test_cpu_features.py \
+            -k 'test_cpu_rdmsr' "
+        ],
+        bkstep.LABEL: "📖 rdmsr",
+        "instances": ["m5d.metal", "m6a.metal", "m6i.metal"],
+    },
+    "cpuid_wrmsr": {
+        "snapshot": {
+            bkstep.COMMAND: [
+                "tools/devtool -y test -- \
+                 -s -ra --nonci  --log-cli-level=INFO \
+                 integration_tests/functional/test_cpu_features.py \
+                 -k 'test_cpu_wrmsr_snapshot or test_cpu_cpuid_snapshot'",
+                "mkdir -pv tests/snapshot_artifacts_upload/{instance}_{os}_{kv}",
+                "sudo mv tests/snapshot_artifacts/* tests/snapshot_artifacts_upload/{instance}_{os}_{kv}"
+            ],
+            bkstep.LABEL: "📸 create snapshots",
+            bkstep.ARTIFACTS: "tests/snapshot_artifacts_upload/**/*",
+            bkstep.TIMEOUT: 30,
+        },
+        "restore": {
+            bkstep.COMMAND: [
+                "buildkite-agent artifact download tests/snapshot_artifacts_upload/{instance}_{os}_{kv}/**/* .",
+                "mv tests/snapshot_artifacts_upload/{instance}_{os}_{kv} tests/snapshot_artifacts",
+                "tools/devtool -y test -- \
+                -s -ra --nonci  --log-cli-level=INFO \
+                integration_tests/functional/test_cpu_features.py \
+                -k 'test_cpu_wrmsr_restore or test_cpu_cpuid_restore'",
+            ],
+            bkstep.LABEL: "📸 load snapshot artifacts from {instance}{snapshot_os}{snapshot_kv} -> {restore_instance}{restore_os}{restore_kv}",
+            bkstep.TIMEOUT: 30,
+        },
+        "snapshot_restore_combinations": [
+            ((("al2", "linux_4.14"), "m5d.metal"), (("al2", "linux_4.14"), "m5d.metal")),
+            ((("al2", "linux_4.14"), "m5d.metal"), (("al2", "linux_4.14"), "m6i.metal")),
+            ((("al2", "linux_4.14"), "m6i.metal"), (("al2", "linux_4.14"), "m5d.metal")),
+            ((("al2", "linux_4.14"), "m6i.metal"), (("al2", "linux_4.14"), "m6i.metal")),
+            ((("al2", "linux_4.14"), "m6a.metal"), (("al2", "linux_4.14"), "m6a.metal")),
+            ((("al2", "linux_5.10"), "m5d.metal"), (("al2", "linux_5.10"), "m5d.metal")),
+            ((("al2", "linux_5.10"), "m5d.metal"), (("al2", "linux_5.10"), "m6i.metal")),
+            ((("al2", "linux_5.10"), "m6i.metal"), (("al2", "linux_5.10"), "m5d.metal")),
+            ((("al2", "linux_5.10"), "m6i.metal"), (("al2", "linux_5.10"), "m6i.metal")),
+            ((("al2", "linux_5.10"), "m6a.metal"), (("al2", "linux_5.10"), "m6a.metal")),
+        ],
```

Comment:
Consider generating this. Maybe keep the data here and generate in group_snapshot_restore

```python
cross_instances = {
  "m5d": ["m6i"],
  "m6i": ["m5d"],
}
src_instances = ["m5d", "m6i", "m6a"]

combs = []
for dp in DEFAULT_PLATFORMS:
    for src_instance in src_instances:
        for dst_instance in [src_instance] + others.get(cross_instances, []):
            combs.append(((dp, src_instance), (dp, dst_instance)))
```